src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/OfflineAuthenticationTest.java

```diff
@@ -154,6 +159,17 @@ private String connectToCore(String thingName, String certificatePem) throws Aut
         }});
     }
 
+    private void corruptStoredClientCertificate(String pem) throws InvalidCertificateException, IOException {
+        NucleusPaths paths = kernel.getNucleusPaths();
+        Path workPath = paths.workPath(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME);
+        Certificate cert = Certificate.fromPem(pem);
+        Path pemFilePath = workPath.resolve("clients").resolve(cert.getCertificateId() + ".pem");
+
+        try (OutputStream writeStream = Files.newOutputStream(pemFilePath)) {
+            writeStream.write("I am evil :)".getBytes());
+        }
```

Comment:
How can we ensure that this is actually corrupted the pem? If we are making assumptions here, we should make sure that the tests will fail is those assumptions don't hold true in the future (e.g. cert store changes the storage format).

Right now the test below will continue to pass

Reply:
I think we might need a cert store fake for this

Reply:
Well we know due to the fact the test fails if I don't ignore the exception. Every time the network goes from up to down and back to up we trigger a background refresh (if we have not run one in the last 24 hours). What this test proofs ids that regardless of what happens to the pem it does not affect the authentication process.

From what we broke on the test scenario
> If Client-Certificate-Store (keystore or the file) gets corrupted (or is unreadable), it does not break normal CDA functioning. (Offline auth should still work, only refresh will be affected).

On thing we can do to improve it is to have some assertions that demonstrate the background refresh failed. We could add an assertion here as well that gets the pem from the registry and tries to build a certificate and fails to check the pem is corrupted

Reply:
Done https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/188/files#diff-7174340e1ac320caaaa268ce778e4e526258177b9409dc6128987d10d6d6905fR409